### PR TITLE
2679702 Remove AngularJS from the NodeJS authorization code sample

### PR DESCRIPTION
**Pull request by Nilesh-Unde**
https://blackbaud.visualstudio.com/Products/_workitems/edit/2679702/

### CODE REVIEW

File: ui/app/main.js [hunk 1]
```diff
@@ -1,82 +1,131 @@
...
+                            // User is authenticated, fetch constituent data
+                            fetch('/api/constituents/280')
+                                .then(response => response.json())
+                                .then(data => {
+                                    const constituent = data;
+                                    renderConstituentData(constituent);
+                                });
+                        }
+                    });
+            }, 200);
```
Comment (Blackbaud-ChristopherRodgers) [inline ui/app/main.js]: 200 ms seems a little too frequent to do this check. This is fine locally. In production, a developer might use something a little more sophisticated that would wait for the `fetch('/auth/authenticated')` response and add a delay before the next check. For demonstration purposes, this is fine. I might go with `500 or 1000`ms here.

File: ui/app/main.js [hunk 2]
```diff
@@ -1,82 +1,131 @@
...
+            const popup = window.open('auth/login?redirect=/%23/auth-success', 'login', 'height=450,width=600');
+            if (window.focus) {
+                popup.focus();
+            }
 
-                if ($window.focus) {
-                    popup.focus();
-                }
-            };
+            let intervalId = setInterval(function () {
```
Comment (Blackbaud-ChristopherRodgers) [inline ui/app/main.js]: This setInterval should also check whether `popup` has already been closed. In testing, I can close the popup and the `fetch('/auth/authenticated')` call will still fire every 200ms. In addition to checking whether the user is authorized, if `popup` is closed, `clearInterval(intervalId)` should also be called.


-----
Bonus: In testing, I can launch multiple popups by clicking the "Authorize using popup" more than once, which will set up multiple redundant `intervals`. Ideally, this page would keep track of the popup and only launch one popup at a time--along with only having one `interval` set at a time. If a popup exists, the page could apply `focus` to that popup the next time the "Authorize using popup" is clicked. However, for demonstration purposes, I don't think this is necessary--only a "nice to have".

File: ui/app/main.js [hunk 3]
```diff
@@ -1,82 +1,131 @@
...
+    }
+
+    // Render constituent data
+    function renderConstituentData(constituent) {
+        const constituentData = `
+        <h1>SKY API Authorization Code Flow Tutorial</h1>
+          <div class="well">
+            <h3 class="well-title">Constituent: ${constituent.name}</h3>
+            <p>
+              See <a href="https://developer.blackbaud.com/skyapi/apis/constituent" target="_blank">Constituent</a>
```
Comment (Blackbaud-ChristopherRodgers) [inline ui/app/main.js]: We can use the updated URL here.
https://developer.blackbaud.com/skyapi/products/renxt/constituent

File: ui/app/main.js (hunk 1 above)
Comment (Nilesh-Unde) [reply ui/app/main.js]: Increased the interval time to 500ms.

File: ui/app/main.js (hunk 2 above)
Comment (Nilesh-Unde) [reply ui/app/main.js]: Added a check to see if the popup is closed before making the API call.

Comment (Nilesh-Unde) [reply ui/app/main.js]: Prevented multiple popups from being opened.

File: ui/app/main.js (hunk 3 above)
Comment (Nilesh-Unde) [reply ui/app/main.js]: Updated the URL

File: ui/app/main.js (hunk 2 above)
Comment (Blackbaud-ChristopherRodgers) [reply ui/app/main.js]: Nice change. Works great!

**Review by Blackbaud-ChristopherRodgers — APPROVED** [review]

**Pull request merged**